src/components/progress/themes/animations.scss

```
@@ -1,37 +1,41 @@
+@use 'sass:map';
 @use 'styles/utilities' as *;
 @use 'vars' as *;
 
-// Linear animation //
-// primary translate
-$indeterminate-primary-translate-step-2: cubic-bezier(.5, 0, .7017, .4958);
-$indeterminate-primary-translate-step-3: cubic-bezier(.3024, .3813, .55, .9563);
+// Animation easing curves
+$easing-curves: (
+    // Primary translate easing curves
+    primary-translate-start: cubic-bezier(0.5, 0, 0.7017, 0.4958),
+    primary-translate-mid: cubic-bezier(0.3024, 0.3813, 0.55, 0.9563),
 
-// primary scale
-$indeterminate-primary-scale-step-2: cubic-bezier(.3347, .124, .7858, 1);
-$indeterminate-primary-scale-step-3: cubic-bezier(.06, .11, .6, 1);
+    // Primary scale easing curves
+    primary-scale-slow-start: cubic-bezier(0.3347, 0.124, 0.7858, 1),
+    primary-scale-quick-end: cubic-bezier(0.06, 0.11, 0.6, 1),
 
-// secondary translate
-$indeterminate-secondary-translate-step-1: cubic-bezier(.15, 0, .515, .4096);
-$indeterminate-secondary-translate-step-2: cubic-bezier(.31, .284, .8, .7337);
-$indeterminate-secondary-translate-step-3: cubic-bezier(.4, .627, .6, .902);
+    // Secondary translate easing curves
+    secondary-translate-start: cubic-bezier(0.15, 0, 0.515, 0.4096),
+    secondary-translate-mid: cubic-bezier(0.31, 0.284, 0.8, 0.7337),
+    secondary-translate-end: cubic-bezier(0.4, 0.627, 0.6, 0.902),
 
-// secondary scale
-$indeterminate-secondary-scale-step-1: cubic-bezier(.15, 0, .515, .4096);
-$indeterminate-secondary-scale-step-2: cubic-bezier(.31, .284, .8, .7337);
-$indeterminate-secondary-scale-step-3: cubic-bezier(.4, .627, .6, .902);
+    // Secondary scale easing curves
+    secondary-scale-slow-start: cubic-bezier(0.15, 0, 0.515, 0.4096),
+    secondary-scale-mid: cubic-bezier(0.31, 0.284, 0.8, 0.7337),
+    secondary-scale-smooth-end: cubic-bezier(0.4, 0.627, 0.6, 0.902)
+);
 
+// Primary animation
 @keyframes indeterminate-primary {
     0% {
         transform: translateX(0);
     }
 
     20% {
-        animation-timing-function: $indeterminate-primary-translate-step-2;
+        animation-timing-function: map.get($easing-curves, primary-translate-start);
```

Wrap this in a function that uses the map.get call internally referencing the $easing-curves. Say the function is named `timing-function`, then your call will simply be:

```suggestion
        animation-timing-function: timing-function(primary-translate-start);
```